modules/compute/src/main/java/org/apache/ignite/internal/compute/executor/ComputeExecutorImpl.java

```diff
@@ -123,6 +131,27 @@ public JobExecutionInternal<ComputeJobDataHolder> executeJob(
         return new JobExecutionInternal<>(execution, isInterrupted, null, false, topologyService.localMember());
     }
 
+    private static Callable<CompletableFuture<ComputeJobDataHolder>> addObservableTimestamp(
+            Callable<CompletableFuture<ComputeJobDataHolder>> jobCallable,
+            ClockService clockService) {
+        return () -> {
+            CompletableFuture<ComputeJobDataHolder> jobFut = jobCallable.call();
+
+            if (jobFut == null) {
+                return CompletableFuture.completedFuture(
+                        new ComputeJobDataHolder(ComputeJobDataType.NATIVE, null, clockService.nowLong()));
+            }
+
+            return jobFut.thenApply(holder -> {
+                if (holder == null) {
+                    return new ComputeJobDataHolder(ComputeJobDataType.NATIVE, null, clockService.nowLong());
+                }
+
+                return new ComputeJobDataHolder(holder.type(), holder.data(), clockService.nowLong());
```

Comment:
Why is nowLong() used here (and twice above) instead of currentLong()?

Reply:
Good catch, should be `currentLong()` as described in the ticket.

Fixed.